cylc/flow/tests/util.py

```diff
@@ -197,13 +201,35 @@ def mock_glbl_cfg(tmp_path, monkeypatch):
 
     * Isolated unit tests within one module.
 
-    # TODO: modify Parsec so we can use StringIO rather than a temp file.
+    Args:
+        pypath (str):
+            The python-like path to the global configuation object you want
+            to fiddle.
+            E.G. if you want to hack the `glbl_cfg` in
+            `cylc.flow.scheduler` you would provide
+            `cylc.flow.scheduler.glbl_cfg`
+        rc_string (str):
+            The globlal configuration as a multi-line string.
+
+    Example:
+        Change the value of `UTC mode` in the global config as seen from
+        `the scheduler` module.
+
+        def test_something(mock_glbl_cfg):
+            mock_glbl_cfg(
+                'cylc.flow.scheduler.glbl_cfg',
+                '''
+                    [cylc]
+                        UTC mode = True
+                '''
+            )
 
     """
-    def _mock(pypath, global_rc):
+    # TODO: modify Parsec so we can use StringIO rather than a temp file.
+    def _mock(pypath, rc_string):
         nonlocal tmp_path, monkeypatch
         tmp_path = tmp_path / 'flow.rc'
```

Comment:
If this method is called more than once it tries to keep creating `flow.rc/flow.rc` etc. Suggestion:
```suggestion
        file_path = tmp_path / 'flow.rc'
```

Reply:
Errr, heck, I'm not sure whether this approach is suitable to being used multiple times in a single test. I think the patching mechanism is liable to break down under that usage.

Reply:
Oh right. I was trying to use it like this, is it an abuse actually? https://github.com/cylc/cylc-flow/blob/4e509e7a121bac25fb120f2d43baec7b248aa158/cylc/flow/tests/test_config.py#L307-L346

Reply:
Nice test.

> is it an abuse actually

If the test works then I guess probably not, will have to research the monkeypatch mechanism.

Reply:
Took a look at `monkeypatch`, it's [painfully simple](https://github.com/pytest-dev/pytest/blob/master/src/_pytest/monkeypatch.py#L171), everything is done in serial, any parallelisation is done in other processes (e.g. via pytest-xdist). Pytest-asyncio might be a curveball but they proclaim it works, how *(or if)* they avoid overlap between monkeypatches from different test functions I don't know.

Reply:
Ok, enabled for re-use and tidied into a conftest file. Good spotting!